### PR TITLE
add the possibility to download sonarqube cli behind proxy

### PR DESCRIPTION
**Pull request by csturiale**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "sonarqube-scanner",
   "description": "SonarQube/SonarCloud Scanner for the JavaScript world",
-  "version": "2.8.2",
+  "version": "2.8.3",
```
Comment (dzienisz) [inline package.json:4]: I think you should not change version

**Review by dzienisz — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (csturiale) [reply package.json:4]: please let me know if you prefer that I rollback the version, I supposed you'd like to release this fix asap as it is forcing teams, working behind proxy, to stick to the previous version

Comment (gabssnake) [reply package.json:4]: We usually do this in a separate commit.

File: src/sonar-scanner-executable.js [hunk 2]
```diff
@@ -100,9 +101,16 @@ function getSonarScannerExecutable(passExecutableCallback) {
   const baseUrl = process.env.SONAR_SCANNER_MIRROR || process.env.npm_config_sonar_scanner_mirror || SONAR_SCANNER_MIRROR
   const fileName = 'sonar-scanner-cli-' + platformBinariesVersion + '-' + targetOS + '.zip'
   const downloadUrl = baseUrl + fileName
+  const proxy = process.env.http_proxy || ''
+  // create an instance of the `HttpsProxyAgent` class with the proxy server information
+  const proxyAgent = new HttpsProxyAgent(proxy)
```
Comment (gabssnake) [inline src/sonar-scanner-executable.js:106]: Not sure we should be creating the instance if there is no proxy value

File: src/sonar-scanner-executable.js [hunk 3]
```diff
@@ -100,9 +101,16 @@ function getSonarScannerExecutable(passExecutableCallback) {
...
   log(`Downloading from ${downloadUrl}`)
   log(`(executable will be saved in cache folder: ${installFolder})`)
-  const downloader = new DownloaderHelper(downloadUrl, installFolder)
+  log(`Using proxy server ${proxy}`)
```
Comment (gabssnake) [inline src/sonar-scanner-executable.js:109]: This log would show a confusing message when no proxy is set.

**Review by gabssnake — CHANGES_REQUESTED** [review]

**Comment by gabssnake** [comment]
I think we would also need to document this in the README, this it was undocumented before.
Could you please include that in the PR?

File: package.json (hunk 1 above)
Comment (csturiale) [reply package.json]: done

File: src/sonar-scanner-executable.js (hunk 2 above)
Comment (csturiale) [reply src/sonar-scanner-executable.js]: I did some changes for the log and safely moved the instance creation inside the condition

File: README.md [hunk 4]
```diff
@@ -135,6 +135,26 @@ or alternatively set variable in `.npmrc`
...
+
+**Behind authenticated proxy:**
+```shell
+export http_proxy=http://user:password@mycompanyproxy.com:PORT
+```
+
+or alternatively set variable in `.npmrc`
+
+```
+    sonar_binary_cache=/Users/myaccount/cache
```
Comment (gabssnake) [inline README.md:155]: Bad copy paste?

**Review by gabssnake — APPROVED** [review]

**Comment by gabssnake** [comment]
There might be something strange with package-lock.json since there are many lines removed?

**Comment by csturiale** [comment]
> There might be something strange with package-lock.json since there are many lines removed?

I suppose that maybe I run npm install using a different Node/npm version

**Review by dzienisz — CHANGES_REQUESTED** [review]

File: package-lock.json [hunk 5]
```diff
@@ -1,5143 +1,8 @@
 {
   "name": "sonarqube-scanner",
   "version": "2.8.2",
-  "lockfileVersion": 2,
+  "lockfileVersion": 1,
```
[the remark this answers is not in the document]
Comment (csturiale) [reply package-lock.json]: done

**Review by dzienisz — APPROVED** [review]

**Pull request merged**